pegasus/helpers/form_helpers.rb

```diff
@@ -135,7 +135,12 @@ def insert_form(kind, data, options={})
     updated_ip: request.ip,
   }
   row[:user_id] = dashboard_user[:id] if dashboard_user
-  row[:id] = DB[:forms].insert(row)
+
+  if kind == "HocSignup2015" && DB[:forms].where(email: row[:email], kind: kind, name: row[:name]).count > 0
```

Comment:
@mehalshah better? we still need the kind check since this function is used for different types of forms